repos/fdbt-site/tests/pages/api/returnService.test.ts

```diff
@@ -0,0 +1,96 @@
+import returnService from '../../../src/pages/api/returnService';
+import {
+    expectedReturnTicketWithAdditionalService,
+    getMockRequestAndResponse,
+    mockRawService,
+} from '../../testData/mockData';
+import * as auroradb from '../../../src/data/auroradb';
+import {
+    MATCHING_JSON_ATTRIBUTE,
+    MATCHING_JSON_META_DATA_ATTRIBUTE,
+    TXC_SOURCE_ATTRIBUTE,
+} from '../../../src/constants/attributes';
+import * as userData from '../../../src/utils/apiUtils/userData';
+
+beforeEach(() => {
+    jest.resetAllMocks();
+
+    jest.spyOn(auroradb, 'getServiceByIdAndDataSource').mockResolvedValue(mockRawService);
+    const s3Spy = jest.spyOn(userData, 'putUserDataInProductsBucketWithFilePath');
+    s3Spy.mockImplementation(() => Promise.resolve('pathToFile'));
+});
+
+describe('returnService', () => {
+    it('should return 302 redirect to /returnService when there is no body in the request', async () => {
+        const writeHeadMock = jest.fn();
+        const { req, res } = getMockRequestAndResponse({
+            cookieValues: {},
+            query: {},
+            body: {
+                selectedServiceId: 1,
+            },
+            uuid: {},
+            mockWriteHeadFn: writeHeadMock,
+        });
+        await returnService(req, res);
+        expect(writeHeadMock).toBeCalledWith(302, {
+            Location: '/returnService?selectedServiceId=1',
+        });
+    });
+
+    it('should return 302 redirect to /returnService when the ticket is not in edit mode', async () => {
```

Comment:
Should it be 302 redirecting back when not in edit mode? I think we should be throwing an error.